src/app/shared/axiosHelper.ts

```diff
@@ -12,13 +13,25 @@ declare module "axios" {
     startTime?: number;
   }
 }
+
 interface RequestLog {
   description: string;
   endpoint: string;
   data?: object;
   cri?: string;
   duration?: number;
 }
+
+// Helper function to determine if an endpoint is in the allow list
+const isEndpointAllowedForDataLogging = (url: string): boolean => {
+  const ALLOWED_ENDPOINTS = Object.values(API_URLS).filter(
+    (endpoint) => endpoint !== API_URLS.API_BUILD_PROVEN_USER_IDENTITY_DETAILS,
```

Comment:
Hmm to me this is still a deny list as we're allowing anything except this request - I think we want to explicitly define the allowed requests? so that if we introduce a new API endpoint in the future, we need to explicitly add it to the allowlist here - otherwise we may start calling a new endpoint that logs PII without us realising